server/src/main/java/org/elasticsearch/cluster/service/TaskBatcher.java

```diff
@@ -120,28 +144,26 @@ private void onTimeoutInternal(List<? extends BatchedTask> tasks, TimeValue time
      */
     protected abstract void onTimeout(List<? extends BatchedTask> tasks, TimeValue timeout);
 
-    void runIfNotProcessed(BatchedTask updateTask) {
+    private void runIfNotProcessed(BatchedTask updateTask) {
         // if this task is already processed, it shouldn't execute other tasks with same batching key that arrived later,
         // to give other tasks with different batching key a chance to execute.
-        if (updateTask.processed.get() == false) {
+        if (!updateTask.processed.get()) {
             final List<BatchedTask> toExecute = new ArrayList<>();
             final Map<String, List<BatchedTask>> processTasksBySource = new HashMap<>();
-            synchronized (tasksPerBatchingKey) {
-                LinkedHashSet<BatchedTask> pending = tasksPerBatchingKey.remove(updateTask.batchingKey);
-                if (pending != null) {
-                    for (BatchedTask task : pending) {
-                        if (task.processed.getAndSet(true) == false) {
-                            logger.trace("will process {}", task);
-                            toExecute.add(task);
-                            processTasksBySource.computeIfAbsent(task.source, s -> new ArrayList<>()).add(task);
-                        } else {
-                            logger.trace("skipping {}, already processed", task);
-                        }
+            final Map<IdentityWrapper, BatchedTask> pending = tasksPerBatchingKey.remove(updateTask.batchingKey);
+            if (pending != null) {
+                for (BatchedTask task : pending.values()) {
+                    if (!task.processed.getAndSet(true)) {
+                        logger.trace("will process {}", task);
+                        toExecute.add(task);
+                        processTasksBySource.computeIfAbsent(task.source, s -> new ArrayList<>()).add(task);
+                    } else {
+                        logger.trace("skipping {}, already processed", task);
                     }
                 }
             }
 
-            if (toExecute.isEmpty() == false) {
+            if (!toExecute.isEmpty()) {
```

Comment:
Please return to the `== false` style and use it everywhere. Reasonable people can disagree on which is preferred between `!` and `== false` but `== false` is the style that we use in our codebase (we do it this way because of the risk `!` being overlooked).

Reply:
I should have followed the established code style, I am sorry. Fixed.